cellengine/resources/population.py

```diff
@@ -1,11 +1,30 @@
 from __future__ import annotations
+from dataclasses import field
+from typing import Optional
+
+from dataclasses_json.cfg import config
+from cellengine.utils.dataclass_mixin import DataClassMixin
+
+from dataclasses import dataclass
 import cellengine as ce
-from cellengine.payloads.population import _Population
 
 
-class Population(_Population):
+@dataclass
+class Population(DataClassMixin):
+
+    _id: str = field(metadata=config(field_name="_id"))
+    name: str
+    experiment_id: str
```

Comment:
I missed this in the attachment PR. Any way to keep the read-only behavior?

Reply:
Yes and no. The best I've been able to figure out so far is:
```python
class ReadOnly:
	def __init__(self):
			self.block_set = False
	def __set_name__(self, owner, attr):
			self.owner = owner.__name__
			self.attr = attr
	def __get__(self, instance, owner):
			return getattr(instance, f"_{self.attr}")
	def __set__(self, instance, value):
			if not self.block_set:
					self.block_set = True
					setattr(instance, f"_{self.attr}", value)
			else:
					raise AttributeError(f"{self.owner}.{self.attr} cannot be set.")

@dataclass
class Foo:
    bar: str = field(default=ReadOnly())
```

But mypy/pyright _really_ doesn't like this

Reply:
I pushed a commit to this branch with read-only fields (only on `experiment_id` for now)

Reply:
#97 proposes a solution to this